src/sessions/__tests__/PageManager.test.ts

```diff
@@ -487,3 +488,97 @@ describe('PageManager tests', () => {
         expect(pageManager.getPage().start).toEqual(2500);
     });
 });
+
+test('when complete referrer is available from the DOM then is recorded in the metadata', async () => {
+    // Init
+    const config: Config = {
+        ...DEFAULT_CONFIG,
+        allowCookies: true
+    };
+    const pageManager: PageManager = new PageManager(config, record);
+
+    Object.defineProperty(document, 'referrer', {
+        value: 'http://abc.com/consoles',
+        configurable: true
+    });
+
+    // Run
+    pageManager.recordPageView('/console/home');
+
+    // Assert
+    expect(pageManager.getPage()).toMatchObject({
+        pageId: '/console/home'
+    });
+
+    expect(pageManager.getAttributes()).toMatchObject({
+        'aws:referrer': 'http://abc.com/consoles',
+        'aws:referrerDomain': 'abc.com'
+    });
+
+    window.removeEventListener(
+        'popstate',
+        (pageManager as any).popstateListener
+    );
+});
+
+test('when only domain level referrer is available from the DOM then is recorded in the metadata', async () => {
+    // Init
+    const config: Config = {
+        ...DEFAULT_CONFIG,
+        allowCookies: true
+    };
+    const pageManager: PageManager = new PageManager(config, record);
+
+    Object.defineProperty(document, 'referrer', {
+        value: 'http://abc.com',
+        configurable: true
+    });
+    // Run
+    pageManager.recordPageView('/console/home');
+
+    // Assert
+    expect(pageManager.getPage()).toMatchObject({
+        pageId: '/console/home'
+    });
+
+    expect(pageManager.getAttributes()).toMatchObject({
+        'aws:referrer': 'http://abc.com',
+        'aws:referrerDomain': 'abc.com'
+    });
+
+    window.removeEventListener(
+        'popstate',
+        (pageManager as any).popstateListener
+    );
+});
+
+test('when referrer from the DOM is empty then it is recorded as empty in the metadata', async () => {
+    // Init
+    const config: Config = {
+        ...DEFAULT_CONFIG,
+        allowCookies: true
+    };
+    const pageManager: PageManager = new PageManager(config, record);
+
+    Object.defineProperty(document, 'referrer', {
+        value: '',
+        configurable: true
+    });
+    // Run
+    pageManager.recordPageView('/console/home');
+
+    // Assert
+    expect(pageManager.getPage()).toMatchObject({
+        pageId: '/console/home'
+    });
+
+    expect(pageManager.getAttributes()).toMatchObject({
+        'aws:referrer': '',
+        'aws:referrerDomain': ''
+    });
```

Comment:
question: do we want to record in metadata if it's empty? is this a conscience decision we're making?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Document/referrer
States that  -`The value is an empty string if the user navigated to the page directly (not through a link, but, for example, by using a bookmark). `, so we should record it